src/telemetry.ts

```diff
@@ -0,0 +1,61 @@
+import * as vscode from "vscode";
+import axios from "axios";
+import {v4 as uuidv4} from 'uuid';
+import * as util from "./util";
+
+const LAST_UPDATED_DATE = "lastUpdatedDate";
+const INSTALLATION_UUID = "installUUID";
+const URL = "https://stats.grafana.org/vscode-extension";
+
+/*
+ * Sends a single anonymous telemetry call once per month, allowing tracking of
+ * usage - reports on first opening of a dashboard each month.
+ */
+export async function sendTelemetry(ctx: vscode.ExtensionContext) {
+
+    const lastUpdatedDate = ctx.globalState.get<string | undefined>(LAST_UPDATED_DATE);
+    const today = new Date();
+
+    if (lastUpdatedDate === undefined) {
+        const uuid = uuidv4();
+        await sendEvent("first", uuid);
+        ctx.globalState.update(LAST_UPDATED_DATE, today);
+        ctx.globalState.update(INSTALLATION_UUID, uuid);
+    } else {
+        if (differentDay(new Date(lastUpdatedDate), today)) {
+            let uuid = ctx.globalState.get(INSTALLATION_UUID);
+            if (uuid === undefined) {
+                console.log("missing uuid");
+                return;
+            }
```

Comment:
```suggestion
            if (uuid === undefined) {
                uuid = uuidv4();
                ctx.globalState.update(INSTALLATION_UUID, uuid);
            }
```

Reply:
This state should never happen though. In which case, perhaps it doesn't matter setting a UUID vs logging an error. Hmm.